docs/fpc-validation-msp.puml

```diff
@@ -0,0 +1,142 @@
+@startuml
+
+'- force wrapping of longer text
+'skinparam wrapWidth 400
+
+!pragma teoz true
+
+hide footbox
+title Foot Box removed
+
+title Fabric Private Chaincode - Transaction validation with MSP
+
+actor "User Org1" as User1 order 9
+
+participant "Peer Org1" as ECC1 order 30
+participant "Peer Org2" as ECC2 order 60
+participant "Peer Org3" as ECC3 order 70
+
+entity Orderer order 100
+
+ref over User1, Orderer
+    install, approve, commit chaincode definition (ECC_MR)
+    Example endorsement policies:
+        - ( 2 of 3 enclaves of different Orgs )
+
+    see fpc-lifecycle-v2.puml
+end ref
+
+note right ECC3 #FFaa00
+    How can we express that we need an enclave here?
+    In particular, an enclave that has ECC_MR and also has a valid attestation?!
+end note
+
+ref over User1, Orderer
+    register enclaves and key-dist
+    see fpc-registration.puml and fpc-key-dist.puml
+end ref
+
+note right Orderer #FFaa00
+    After discussion with Angelo with have two options:
+
+    1) MSP integration via explicit chaincode endorsement policies
+        where we first register all enclaves and then update
+        the chaincode policy and state explicitly which enclaves
+        can endorse.
+
+    2) We create a new MPS principle which, when presents,
+        triggers the peer to check the attestation report attached
+        to an enclave x509 cert.
+
+    Detailed evaluation of those two approaches is needed.
+end note
+
+group endorsement
+    activate User1
+
+    User1 -> ECC1 ++: send proposal
+    ref over ECC1
+        invoke chaincode enclave
+        see fpc-cc-invocation.puml
+    end ref
+    return proposal response
+
+    User1 -> ECC2 ++: send proposal
+    ref over ECC2
+        invoke chaincode enclave
+        see fpc-cc-invocation.puml
+    end ref
+    return
+
+    User1 -> User1 : form transaction
+
+    User1 -> Orderer++ : send transaction
+    deactivate User1
+
+end group
+
+    group collect transactions
+        ...
+        Orderer -> Orderer : form block
+    end group
+
+    Orderer -> ECC1+ : broadcast
+    & Orderer -> ECC2+ : broadcast
+    & Orderer -> ECC3+ : broadcast
+    deactivate Orderer
+
+    group validation & commit
+        ECC1 -> ECC1 : validate block signature
+        loop for each tx in block
+            ECC1 -> ECC1 : validate tx (client) signature
+
+            ECC1 -> ECC1 : validate endorsements policy
+            note right #FFaa00
+                - is endorsing enclave part of Org_i?
+                - has enclave a proper attestation?
+                - is this enclave also registered to
```

Comment:
here we should not overload MSP with operations it is not supposed to do, so checking the registration should be actually cast as membership check, where the membership is maintained by MSP itself.